includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1055,6 +1056,26 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 		];
 	}
 
+	/**
+	 * Given the charge data, checks if there was an exchange and adds it to the given order as metadata
+	 *
+	 * @param WC_Order $order The order to update.
+	 * @param string   $charge_id ID of the charge to attach data from.
+	 */
+	public function attach_exchange_info_to_order( $order, $charge_id ) {
+		$currency_order   = $order->get_currency();
+		$currency_account = $this->account->get_account_default_currency();
+
+		if ( strtolower( $currency_order ) !== $currency_account ) {
+			// We check that the currency used in the order is different than the one set in the WC Payments account
+			// to avoid requesting the charge if not needed.
+			$charge        = $this->payments_api_client->get_charge( $charge_id );
+			$exchange_rate = $charge['balance_transaction']['exchange_rate'];
+			$order->update_meta_data( '_wcpay_multi_currency_stripe_exchange_rate', $exchange_rate );
```

Comment:
Can we add an additional check here to make sure that the currency of `$charge` matches the store default currency? 

I think we could just do something like:

```php
if ( $charge['balance_transaction']['currency'] === strtolower( get_woocommerce_currency() ) ) {
    $order->update_meta_data( '_wcpay_multi_currency_stripe_exchange_rate', $exchange_rate );
    $order->save_meta_data();
}
```

The reason for this is I want to double check that the exchange rate is actually the rate for a conversion between the customer currency and the WooCommerce default currency. There may for example, be cases where the customer has multiple currency bank accounts for different currencies in Stripe, so a conversion wasn't actually required, and in those cases we wouldn't want to save a rate.

Reply:
Would checking the order currency against the `$account['store_currencies']['supported']` help in this case? IIRC that prop should contain every external account's currency that shouldn't require conversion when creating the payment intent.

Reply:
@luizreis I chatted to David on slack about this and we think this can be solved by just checking the `$charge['balance_transaction']['exchange_rate']` before saving the rate. If no conversion was performed, it'll just be null, so we can just not save it in that case. Does that make sense?

Reply:
It does. The main difference would be that we could avoid fetching the charge details if checking against the external account's currencies. This approach you mentioned is simpler and probably more stable than the one I outlined, though, so I'd say :shipit: 😄